cmd/service-controller/controller.go

```diff
@@ -313,11 +329,21 @@ func (c *Controller) checkServiceFor(desired *ServiceBindings, actual *corev1.Se
 		}
 		if actual.Spec.Ports[0].TargetPort.IntValue() != desired.ingressPort {
 			update = true
+			originalAssignedPort, _ := strconv.Atoi(actual.Annotations[types.OriginalAssignedQualifier])
+			actualTargetPort := actual.Spec.Ports[0].TargetPort.IntValue()
+			// If target port has been modified by user
+			if actualTargetPort != originalAssignedPort {
+				actual.ObjectMeta.Annotations[types.OriginalTargetPortQualifier] = strconv.Itoa(actualTargetPort)
+			}
+			actual.ObjectMeta.Annotations[types.OriginalAssignedQualifier] = strconv.Itoa(desired.ingressPort)
 			actual.Spec.Ports[0].TargetPort = intstr.FromInt(desired.ingressPort)
 		}
 	}
 	if desired.headless == nil && !equivalentSelectors(actual.Spec.Selector, kube.GetLabelsForRouter()) {
 		update = true
+		if !hasOriginalSelector(*actual) || !hasRouterSelector(*actual) {
```

Comment:
The preceding if statement will only be true if the selector is not the router selector. So is there any value in the additional test for !hasRouterSelector() here? Also why would we not update the originalSelector annotation even if it already exists, since presumably the selector has somehow been changed again by the user in that case?

Reply:
That is right and odd (not sure why I left it there). I will remove and double-check.
Thanks for pointing it out.